### ISSUE TITLE
Define the behavior of AvoidAsyncVoid for event handlers

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/AsyncUsageAnalyzers#15]
Currently the **AvoidAsyncVoid** analyzer does not have any special consideration for event handlers. The desired behavior for these cases should be explicitly documented, even if the implementation is kept as-is.

**Comment by paulomorgado** [comment]
@sharwell, I don't understand what you mean. Do you mean that this should be reported for event handlers and explicitly ignored?

**Comment by sharwell** [comment]
@paulomorgado I'm saying we should do one of the following:

1. Keep the code as-is, but add documentation
2. Modify the code so async void event handlers are allowed *and* document it

**Comment by JohanLarsson** [comment]
Related: https://github.com/DotNetAnalyzers/AsyncUsageAnalyzers/issues/53

**Comment by paulomorgado** [comment]
@sharwell, I'm sorry but I'm still not getting it.

By "modify the code" do you mean adding `#pragma` directives?

**Comment by sharwell** [comment]
@paulomorgado No, I mean modify the way this analyzer examines code. Right now there is no special analysis for event handlers so they are analyzed the same way as any other method.

**Comment by paulomorgado** [comment]
Oh! I thought you're saying something else.
